### DISCUSSION TITLE
Some routes are localized, some not, best way to proceed ?

### DISCUSSION THREAD

**Discussion by ScreamZ** [amannn/next-intl#1031] [Q&A] [answered]
> [!NOTE]
> Some solution with [route group](https://nextjs.org/docs/app/building-your-application/routing/route-groups#creating-multiple-root-layouts) below in the section (see EDIT)

Hello,

I have some routes using localization patterns while some other doesn't.

```
/fr/xxx
/en/xxx
/xyz -> not using
```

Currently in `app/[locale]/layout.tsx` i have this that define the `html` language

```
<html lang={locale}>
    <body className={archivo.variable}>{children}</body>
</html>
```

How can I proceed to set my default html lang in the root layout `app/layout.tsx` ? I cannot use `usePathName` as it would make my layout client only.

Currently my `app/layout.tsx` is as follow to support my `not-found` global as defined in the doc.
```tsx
// Since we have a root `not-found.tsx` page, a layout file
// is required, even if it's just passing children through.
export default function RootLayout({ children }: { children: React.ReactNode }) {
  return children;
}
```

and my not found is 

```tsx
"use client";

import Error from "next/error";

export default function NotFound() {
  return (
    <html lang="en">
      <body>
        <Error statusCode={404} />
      </body>
    </html>
  );
}
```

Do I need to add <html> like this on all my pages? not part of the lang layout?

Thanks

## EDIT
I've managed to better organize, which seems to be solving my problem.

I've created a route group `not-localized` that allows me to define a new `layout.tsx` and in this one I'm doing my job

<img width="334" alt="Screenshot 2024-04-29 at 18 13 22" src="https://github.com/amannn/next-intl/assets/6640835/59b8e806-b024-43d9-8210-9cb61403e8d3">

Also updated my `middleware.ts` like so
```diff
import { AvailableLocales } from "i18n/locales";
import createMiddleware from "next-intl/middleware";

export default createMiddleware({
  locales: AvailableLocales,
  defaultLocale: AvailableLocales[0], // As `localeDetection` is it will try guess the locale from the incoming request, this is the last fallback
  localeDetection: true,
});

export const config = {
  // Match only internationalized pathnames, this cannot be used with code so anytime a locale is used, the locale must be hardcoded here
-        matcher: ["/", `/(en|fr)/:path*`],
+        matcher: [`/(en|fr)/:path*`],
};
```

**Comment by amannn** [comment, marked as answer]
That's an interesting solution with the route group, thanks for sharing it here!
